src/main/java/mops/termine2/controller/TermineNeuController.java

```diff
@@ -166,72 +142,34 @@ public String terminLoeschen(Principal principal, Model model,
-		LocalDateTime minVorschlag = LocalDateTimeManager.bekommeFruehestesDatum(gueltigeVorschlaege);
-		LocalDateTime maxVorschlag = LocalDateTimeManager.bekommeSpaetestesDatum(gueltigeVorschlaege);
-		
-		if (gueltigeVorschlaege.isEmpty()) {
-			gueltigeVorschlaege.add(null);
-			fehler = "Es muss mindestens einen Vorschlag geben.";
-		}
-		
-		// minVorschlag and maxVorschlag are always null together
-		if (minVorschlag != null) {
-			terminfindungService.setzeFrist(terminfindung, minVorschlag);			
-			terminfindungService.setzeLoeschdatum(terminfindung, maxVorschlag);
-		}
-		
-		if (LocalDateTimeManager.istVergangen(terminfindung.getFrist().minusMinutes(5))) {
-			fehler = "Die Frist ist zu kurzfristig.";
-		}
-		
-		terminfindung.setVorschlaege(gueltigeVorschlaege);
 		
-		// Terminfindung erstellen
-		terminfindung.setErsteller(account.getName());
+		List<String> fehler = terminfindungService.erstelleTerminfindung(account,
+			terminfindung);
+		fehler.addAll(linkService.setzeLink(terminfindung));
+		gruppeService.setzeGruppeId(terminfindung, gruppeSelektiert);
 		
-		if (gruppeSelektiert.getId() != null && !gruppeSelektiert.getId().equals("-1")) {
-			Gruppe gruppe = gruppeService.loadByGruppeId(gruppeSelektiert.getId());
-			terminfindung.setGruppeId(gruppe.getId());
-		}
-		
-		if (terminfindung.getLink().isEmpty()) {
-			String link = linkService.generiereEindeutigenLink();
-			terminfindung.setLink(link);
-		} else {
-			if (!linkService.pruefeEindeutigkeitLink(terminfindung.getLink())) {
-				fehler = "Der eingegebene Link existiert bereits.";
-			}
-			if (!linkService.isLinkValid(terminfindung.getLink())) {
-				fehler = "Der eingegebene Link enthält ungültige Zeichen";
-			}
-		}
-		
-		if (!fehler.equals("")) {
-			model.addAttribute("gruppen", gruppeService.loadByBenutzer(account));
-			model.addAttribute("gruppeSelektiert", gruppeSelektiert);
-			model.addAttribute("terminfindung", terminfindung);
-			model.addAttribute("fehler", fehler);
+		if (fehler.isEmpty()) {
+			terminfindungService.save(terminfindung);
+			logger.info("Benutzer '" + account.getName() + "' hat eine neue Terminabstimmung mit Link '"
+				+ terminfindung.getLink() + "' erstellt");
 			
-			return "termine-neu";
+			redirectAttributes.addFlashAttribute("erfolg", "Der Termin wurde gespeichert.");
+			return "redirect:/termine2";
 		}
 		
-		terminfindungService.save(terminfindung);
-		logger.info("Benutzer '" + account.getName() + "' hat eine neue Terminabstimmung mit link '"
-			+ terminfindung.getLink() + "' erstellt");
+		model.addAttribute(Konstanten.MODEL_ACCOUNT, account);
+		model.addAttribute(Konstanten.MODEL_GRUPPEN, gruppeService.loadByBenutzer(account));
+		model.addAttribute(Konstanten.MODEL_GRUPPE_SELEKTIERT, gruppeSelektiert);
+		model.addAttribute(Konstanten.MODEL_TERMINFINDUNG, terminfindung);
+		model.addAttribute(Konstanten.MODEL_FEHLER, fehler.get(fehler.size() - 1));
```

Comment:
Wenn wir schon alle Fehler in einer Liste haben, sollten wir diese vielleicht auch alle an das UI übergeben und anzeigen.

Reply:
Das mit der Liste habe ich Vorbereitung darauf gemacht. Ich würde das an das UI Team weitergeben, damit das UI mit der Liste umgehen kann. Ich habe das so gemacht, um alles kompatibel zu halten.